### DISCUSSION TITLE
Teams feature: Workflow restructure

### DISCUSSION THREAD

**Discussion by ben-hearn-sb** [pretix/pretix#2262] [Ideas]
Hello!

We are now using the 4.4.0.dev0 version as we wanted to make use of the teams feature for our organisation.

The feature is a great idea but the workflow could benefit from a bit of restructuring:

The add button & email address in this workflow are the main issues. An add button in a row based structure indicates adding a new row since we are in a row context already.

When I first tried to add a new user I pressed this thinking this is how I should add a new user row and I got the error message that the changes could not be saved. This is a confusing workflow and a confusing error message, it took me few minutes to muddle through to find out how it worked.

**Email address issues:**
Pretix gives the ability already to choose from existing organisations when creating an event, could it not also auto drop down all users so you could quickly find someone instead of having to go into the users page and copy/paste their email address.

![image](https://user-images.githubusercontent.com/13691166/138566229-6d10486d-c1e9-4c6b-b8a1-9a941fd0563a.png)

To make the whole workflow more obvious I propose the following restructure:

Add clicked -> New row is added -> Email typed in or chosen from drop down of existing users -> Save button to save team changes
- Overall save team button will give finality to the changes on the team
- Is inline with all other settings dialogs in the system
- Saving the team button will save the current pretix user to your team or send an email to a new team member not in the system

**The new layout would ideally look like this:**

![image](https://user-images.githubusercontent.com/13691166/138566975-c9551034-b5bd-4419-b391-def6f71113e9.png)

![image](https://user-images.githubusercontent.com/13691166/138567021-ffea3059-2ae6-4ae6-9ab9-86b8db8a2a98.png)

Anyway, that's my 2cents for this great new feature!

// Ben

**Comment by raphaelm** [comment]
Thank you a lot on your very valuable feedback!

(First of all, I'm not sure where the impression comes from that this a new feature, teams have been introduced in May **2017** and have barely changed since… If you haven't upgraded pretix since pre-1.4 that might just be a new record :wink:)

A dropdown of existing email addresses is not going to happen, as it would be a major privacy and security issue! Being manager of a team does not mean you are allowed to see all accounts on the entire installation of pretix. For our pretix Hosted installation e.g. this would mean access to ten thousands of user email addresses.

I think you're feedback on the "Add" button is valid, having it behave like you suggest would be a slight improvement in making it more consistent with other UIs (at the same time, it would introduce a JavaScript dependency for that page and more clicks, but I think that'd be fine). I don't think it's a high priority though, since in the last 4 years I do not remember a single customer on the phone not figuring out the button very quickly ;)

**Reply by ben-hearn-sb** [reply]
As per the email addresses, Ok I see. I was talking in a more local sense since or organisation hosts our own server. I had forgotten that there is also a very large scaled public version of pretix.

Teams was from 2017? Interesting, we upgraded to 4.4.0.dev0 as we could only see teams documentation attached to that particular build.

Its not a high prio issue I agree and once you have figured it out it's all good. As per customers not figuring it out, most customers are not tools engineers :P 

Anyway, thanks for the response, pretix is great!
